fuel-tx/src/transaction.rs

```diff
@@ -551,20 +547,48 @@ pub mod field {
         Word,
     };
 
+    use crate::policies::PolicyType;
     use alloc::vec::Vec;
     use core::ops::{
         Deref,
         DerefMut,
     };
 
     pub trait GasPrice {
-        fn gas_price(&self) -> &Word;
-        fn gas_price_mut(&mut self) -> &mut Word;
-        fn gas_price_offset(&self) -> usize {
-            Self::gas_price_offset_static()
+        fn gas_price(&self) -> Word;
+        fn set_gas_price(&mut self, value: Word);
+    }
+
+    impl<T: Policies + ?Sized> GasPrice for T {
```

Comment:
Curious why you include `?Sized` in all the blanket impls.

Reply:
By default in Rust, the implementation requires `Sized`, while the trait definition - no. Because implementation of the `Policies` trait is enough to have the other four traits implementations (`GasPrice`, `WitnessLimit`, `Maturity` `MaxFeeLimit`), I don't want to require `Sized` on the `Policies` or in any other bounds which try to utilize this blanket implementation.

You can try to remove `?Sized` and you will see that I'm talking about=)

Reply:
Hmm. Okay. At a high level I understand what it's saying.

> You can try to remove ?Sized and you will see that I'm talking about=)

It still compiles without it. So I guess I was wondering if it's a "best practice" sort of thing or if it is required by something depending on the trait.

I'm not worried about it, just trying to learn.